src/wordlift/post-excerpt/class-post-excerpt-meta-box-adapter.php

```diff
@@ -0,0 +1,128 @@
+	private $wordpress_excerpt_callback;
+
+	public function __construct() {
+		$this->wordpress_excerpt_callback = NULL;
+	}
+
+	/**
+	 * Removes the registered post excerpt metabox.
+	 */
+	private function remove_default_post_excerpt_meta_box() {
+		remove_meta_box( self::POST_EXCERPT_META_BOX_KEY, get_current_screen(), 'normal' );
+	}
+
+	/**
+	 * Adds the custom post excerpt metabox.
+	 */
+	private function add_custom_post_excerpt_meta_box() {
+		add_meta_box(
+			self::POST_EXCERPT_META_BOX_KEY,
+			__( 'Post Excerpt', 'wordlift' ),
+			array( $this, 'print_wordlift_custom_post_excerpt_box' )
+		);
+	}
+
+	public function get_post_excerpt_translations() {
+		return array(
+			'orText' => 'Or use wordlift suggested post excerpt',
+            'generatingText' => 'Generating excerpt...',
+			'restUrl'                 => get_rest_url( NULL, WL_REST_ROUTE_DEFAULT_NAMESPACE . '/post-excerpt' ),
+			'nonce'                   => wp_create_nonce( 'wp_rest' ),
+			'postId'                  => get_the_ID(),
+		);
+	}
+
+	private function print_wordpress_post_excerpt_ui($post) {
+		?>
+		<label class="screen-reader-text" for="excerpt">
+			<?php _e( 'Excerpt' ); ?>
+		</label>
+		<textarea rows="1" cols="40" name="excerpt" id="excerpt">
+			<?php echo $post->post_excerpt; // textarea_escaped ?>
+		</textarea>
+		<?php
+	}
+
+	private function print_wordpress_post_excerpt_custom_text() {
+		?>
+		<p>
+			<?php
+			printf(
+			/* translators: %s: Documentation URL. */
+				__( 'Excerpts are optional hand-crafted summaries of your content that can be used in your theme. <a href="%s">Learn more about manual excerpts</a>.' ),
+				__( 'https://wordpress.org/support/article/excerpt/' )
+			);
+			?>
+		</p>
+		<?php
+	}
+
+	public function print_wordlift_custom_post_excerpt_box() {
+		global $post;
+		// Print the wordpress template for post excerpt
+		$this->print_wordpress_post_excerpt_ui($post);
+		// Invoke our call back to add additional html, the react script will find this id and render the component there.
+		echo "<div id='" . self::WORDLIFT_EXCERPT_DIV_ID . "'></div>";
+		$this->print_wordpress_post_excerpt_custom_text();
+	}
+
+	private function enqueue_post_excerpt_scripts() {
+		Scripts_Helper::enqueue_based_on_wordpress_version(
+			'wl-post-excerpt',
+			plugin_dir_url( dirname( dirname( __FILE__ ) ) ) . 'js/dist/post-excerpt',
+			array( 'react', 'react-dom', 'wp-polyfill' ),
+			TRUE
+		);
+		wp_enqueue_style(
+			'wl-post-excerpt-style',
+			plugin_dir_url( dirname( dirname( __FILE__ ) ) ) . 'js/dist/post-excerpt.css',
+			array()
+		);
+		wp_localize_script( 'wl-post-excerpt',
+			'_wlExcerptSettings',
+			$this->get_post_excerpt_translations() );
+	}
+
+	/**
+	 * Replaces the default post excerpt meta box with custom post excerpt meta box.
+	 */
+	public function replace_post_excerpt_meta_box() {
+		$this->remove_default_post_excerpt_meta_box();
+		$this->add_custom_post_excerpt_meta_box();
+		$this->enqueue_post_excerpt_scripts();
+	}
```

Comment:
@naveen17797 in Classic Editor edit screen, this level of hijacking looks too extreme to me as we're completely rewriting WP's metabox. Maybe it's better to inject ourselves client-side using JS?